File: php/features.php
<?php
declare(strict_types=1);

// MagicTheGatheringTwo SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class MagicTheGatheringTwoFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new MagicTheGatheringTwoBaseFeature();
            case "test":
                return new MagicTheGatheringTwoTestFeature();
            default:
                return new MagicTheGatheringTwoBaseFeature();
        }
    }
}

File: py/features.py
# MagicTheGatheringTwo SDK feature factory

from feature.base_feature import MagicTheGatheringTwoBaseFeature
from feature.test_feature import MagicTheGatheringTwoTestFeature


def _make_feature(name):
    features = {
        "base": lambda: MagicTheGatheringTwoBaseFeature(),
        "test": lambda: MagicTheGatheringTwoTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
